asapdiscovery-dataviz/asapdiscovery/dataviz/viz_v2/html_viz.py

```diff
@@ -0,0 +1,97 @@
+from enum import Enum
+from pathlib import Path
+
+from asapdiscovery.data.fitness import target_has_fitness_data
+from asapdiscovery.data.postera.manifold_data_validation import TargetTags
+from asapdiscovery.dataviz.html_viz import HTMLVisualizer
+from asapdiscovery.dataviz.viz_v2.visualizer import VisualizerBase
+from asapdiscovery.docking.docking_data_validation import (
+    DockingResultColsV2 as DockingResultCols,
+)
+from asapdiscovery.docking.docking_v2 import DockingResult
+from pydantic import Field, root_validator
+
+
+class ColourMethod(str, Enum):
+    subpockets = "subpockets"
+    fitness = "fitness"
+
+
+class HTMLVisualizerV2(VisualizerBase):
+    """
+    Class for generating HTML visualizations of poses.
+    """
+
+    target: TargetTags = Field(..., description="Target to visualize poses for")
+    colour_method: ColourMethod = Field(
+        ColourMethod.subpockets,
+        description="Protein surface coloring method. Can be either by `subpockets` or `fitness`",
+    )
+    debug: bool = Field(False, description="Whether to run in debug mode")
+    output_dir: Path = Field(
+        "html", description="Output directory to write HTML files to"
+    )
+
+    @root_validator
+    @classmethod
+    def must_have_fitness_data(cls, values):
+        target = values.get("target")
+        colour_method = values.get("colour_method")
+        if colour_method == ColourMethod.fitness and not target_has_fitness_data(
+            target
+        ):
+            raise ValueError(
+                f"Attempting to colour by fitness and {target} does not have fitness data, use `subpockets` instead."
+            )
+        return values
+
+    def get_tag_for_colour_method(self):
+        """
+        Get the tag to use for the colour method.
+        """
+        if self.colour_method == ColourMethod.subpockets:
+            return DockingResultCols.HTML_PATH_POSE.value
+        elif self.colour_method == ColourMethod.fitness:
+            return DockingResultCols.HTML_PATH_FITNESS.value
+
+    def _visualize(self, docking_results: list[DockingResult]) -> list[dict[str, str]]:
+        """
+        Visualize a list of docking results.
+
+        NOTE: This is an extremely bad way of doing this, but it's a quick fix for now
+        """
+        data = []
+        for result in docking_results:
+            # sorryyyyy
+            output_pref = result.get_combined_id()
+            outpath = self.output_dir / output_pref / "pose.html"
+            viz_class = HTMLVisualizer(
+                [result.posed_ligand.to_oemol()],
+                [outpath],
+                self.target,
+                result.to_protein(),
+                self.colour_method,
+                logger=None,
+                debug=self.debug,
+            )
```

Comment:
this is the part that will be fixed by #617 ?

Reply:
Yep, a bit gross atm, but we will straighten out.